x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.ts

```diff
@@ -508,32 +506,52 @@ export class SecureSavedObjectsClientWrapper implements SavedObjectsClientContra
     type: string | string[],
     options: SavedObjectsOpenPointInTimeOptions
   ) {
-    try {
-      const args = { type, options };
-      await this.legacyEnsureAuthorized(type, 'open_point_in_time', options?.namespace, {
+    const args = { type, options };
+    const { status, typeMap } = await this.legacyEnsureAuthorized(
+      type,
+      'open_point_in_time',
+      options?.namespaces,
+      {
         args,
         // Partial authorization is acceptable in this case because this method is only designed
         // to be used with `find`, which already allows for partial authorization.
         requireFullAuthorization: false,
-      });
-    } catch (error) {
+      }
+    );
+
+    if (status === 'unauthorized') {
       this.auditLogger.log(
         savedObjectEvent({
           action: SavedObjectAction.OPEN_POINT_IN_TIME,
-          error,
+          error: new Error(status),
         })
       );
-      throw error;
+      throw SavedObjectsErrorHelpers.decorateForbiddenError(new Error(status));
     }
 
+    const typeToNamespacesMap = Array.from(typeMap).reduce<Map<string, string[] | undefined>>(
+      (acc, [currentType, { authorizedSpaces, isGloballyAuthorized }]) =>
+        isGloballyAuthorized
+          ? acc.set(currentType, options.namespaces)
+          : acc.set(currentType, authorizedSpaces),
+      new Map()
+    );
+
     this.auditLogger.log(
       savedObjectEvent({
         action: SavedObjectAction.OPEN_POINT_IN_TIME,
         outcome: 'unknown',
       })
     );
 
-    return await this.baseClient.openPointInTimeForType(type, options);
+    return await this.baseClient.openPointInTimeForType(
+      status === 'partially_authorized' ? '' : type, // if the user is partially authorized, type must be an empty string to use typeToNamespacesMap instead
+      {
+        ...options,
+        typeToNamespacesMap: undefined, // if the user is fully authorized, use `undefined` as the typeToNamespacesMap to prevent privilege escalation
+        ...(status === 'partially_authorized' && { typeToNamespacesMap, namespaces: [] }), // the repository requires that `type` and `namespaces` must be empty if `typeToNamespacesMap` is defined
+      }
+    );
```

Comment:
Per my top level review comment, I'd suggest you change this:
```diff
--- a/x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.ts
+++ b/x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.ts
@@ -529,14 +529,6 @@ export class SecureSavedObjectsClientWrapper implements SavedObjectsClientContra
       throw SavedObjectsErrorHelpers.decorateForbiddenError(new Error(status));
     }
 
-    const typeToNamespacesMap = Array.from(typeMap).reduce<Map<string, string[] | undefined>>(
-      (acc, [currentType, { authorizedSpaces, isGloballyAuthorized }]) =>
-        isGloballyAuthorized
-          ? acc.set(currentType, options.namespaces)
-          : acc.set(currentType, authorizedSpaces),
-      new Map()
-    );
-
     this.auditLogger.log(
       savedObjectEvent({
         action: SavedObjectAction.OPEN_POINT_IN_TIME,
@@ -544,14 +536,8 @@ export class SecureSavedObjectsClientWrapper implements SavedObjectsClientContra
       })
     );
 
-    return await this.baseClient.openPointInTimeForType(
-      status === 'partially_authorized' ? '' : type, // if the user is partially authorized, type must be an empty string to use typeToNamespacesMap instead
-      {
-        ...options,
-        typeToNamespacesMap: undefined, // if the user is fully authorized, use `undefined` as the typeToNamespacesMap to prevent privilege escalation
-        ...(status === 'partially_authorized' && { typeToNamespacesMap, namespaces: [] }), // the repository requires that `type` and `namespaces` must be empty if `typeToNamespacesMap` is defined
-      }
-    );
+    const allowedTypes = [...typeMap.keys()]; // only allow the user to open a PIT against indices for type(s) they are authorized to access
+    return await this.baseClient.openPointInTimeForType(allowedTypes, options);
   }
 
   public async closePointInTime(id: string, options?: SavedObjectsClosePointInTimeOptions) {
```

and remove the changes to this function in the SOR 😄

Reply:
Yea, that's definitely better. Thanks!